sm/plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -48,6 +60,20 @@ impl InternalError {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum UpdateAction {
+    Install,
+    Remove,
+}
+#[derive(Debug, Deserialize)]
+struct SoftwareModuleUpdate {
+    pub action: UpdateAction,
+    pub name: String,
+    pub version: Option<String>,
+    pub path: Option<String>,
```

Comment:
To parse not present arguments there is the default attribute which you can apply to optional fields, something like this:
```suggestion
    #[serde(default)]
    pub version: Option<String>,
    #[serde(default)]
    pub path: Option<String>,
```